experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/proxies.spec.ts

```diff
@@ -201,6 +201,24 @@ describe("SharedTreeList", () => {
 			const schema = _.intoSchema(_.list(_.number));
 			const initialTree = [0, 1, 2, 3];
 
+			itWithRoot("moveToStart()", schema, initialTree, (list) => {
+				assert.deepEqual(list, [0, 1, 2, 3]);
+				list.moveToStart(1);
+				assert.deepEqual(list, [1, 0, 2, 3]);
+			});
+
+			itWithRoot("moveToEnd()", schema, initialTree, (list) => {
+				assert.deepEqual(list, [0, 1, 2, 3]);
+				list.moveToEnd(1);
+				assert.deepEqual(list, [0, 2, 3, 1]);
+			});
+
+			itWithRoot("moveToIndex()", schema, initialTree, (list) => {
+				assert.deepEqual(list, [0, 1, 2, 3]);
+				list.moveToIndex(1, 2);
+				assert.deepEqual(list, [0, 2, 1, 3]);
+			});
```

Comment:
Would be good to test a rightward move as well. To validate the semantics of the destination index being interpreted after the detaching of the moved node.

Reply:
Done, and thank you. Daniel recently changed the rightward move behavior and we forgot to update the docs, so this prompted me to.